### PR TITLE
sqlc 1.6.0 (new formula)

### PR DESCRIPTION
**Pull request by kyleconroy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Hey there Homebrew team. I'm the maintainer of [sqlc](https://github.com/kyleconroy/sqlc) and I'd love for it to be included in core. I've maintained a tap for the last year (https://github.com/kyleconroy/homebrew-sqlc), but it's not the same as `brew install sqlc`.

Thanks!

### CODE REVIEW

File: Formula/sqlc.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  homepage "https://sqlc.dev/"
+  url "https://github.com/kyleconroy/sqlc/archive/v1.6.0.tar.gz"
+  sha256 "a95a123f29a71f5a2eea0811e4590d59cbc92eccd407abe93c110c738fc4740b"
+  license "MIT"
+  head "https://github.com/kyleconroy/sqlc.git"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-trimpath", "-o", bin/"sqlc", "./cmd/sqlc"
```
Comment (dawidd6) [inline Formula/sqlc.rb:12]:
```suggestion
    system "go", "build", *std_go_args, "./cmd/sqlc"
```

**Review by chenrui333 — DISMISSED** [review]

File: Formula/sqlc.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    SQLC
+
+    (testpath/"query.sql").write <<~SQL.squish
+      CREATE TABLE foo (bar text);
+
+      -- name: SelectFoo :many
+      SELECT * FROM foo;
+    SQL
+
+    system bin/"sqlc", "generate"
```
Comment (chenrui333) [inline Formula/sqlc.rb:38]: it would be nice to assert the response like other go formulae are doing.

Comment (kyleconroy) [reply Formula/sqlc.rb:38]: @chenrui333 do you have an example I can look at?

Comment (gromgit) [reply Formula/sqlc.rb:38]: @kyleconroy See the `mysql` test block, for instance. Relevant docs:

* https://www.rubydoc.info/github/test-unit/test-unit/Test/Unit/Assertions
* https://rubydoc.brew.sh/Homebrew/Assertions.html#shell_output-instance_method

Comment (chenrui333) [reply Formula/sqlc.rb:38]: I wrote some tests, should be good to go now :)

Comment (kyleconroy) [reply Formula/sqlc.rb:38]: Can you fix the one lint error?

File: Formula/sqlc.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  homepage "https://sqlc.dev/"
+  url "https://github.com/kyleconroy/sqlc/archive/v1.6.0.tar.gz"
+  sha256 "a95a123f29a71f5a2eea0811e4590d59cbc92eccd407abe93c110c738fc4740b"
+  license "MIT"
+  head "https://github.com/kyleconroy/sqlc.git"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "./cmd/sqlc"
```
Comment (bayandin) [inline Formula/sqlc.rb:12]:
```suggestion
    system "go", "build", *std_go_args, "-ldflags", "-s -w", "./cmd/sqlc"
```

Comment (kyleconroy) [reply Formula/sqlc.rb:12]: Can you explain why these flags should be added?

Comment (bayandin) [reply Formula/sqlc.rb:12]: It produces binaries of smaller sile (https://blog.filippo.io/shrink-your-go-binaries-with-this-one-weird-trick/):
``` 
# go build -trimpath -o /usr/local/Cellar/sqlc/1.6.0/bin/sqlc ./cmd/sqlc
$ du -h /usr/local/Cellar/sqlc/1.6.0/bin/sqlc 
 37M	/usr/local/Cellar/sqlc/1.6.0/bin/sqlc
```
```
# go build -trimpath -o /usr/local/Cellar/sqlc/1.6.0/bin/sqlc -ldflags -s -w ./cmd/sqlc
$ du -h /usr/local/Cellar/sqlc/1.6.0/bin/sqlc 
 28M	/usr/local/Cellar/sqlc/1.6.0/bin/sqlc
```

And most of `go`-formulae in homebrew-core are using these ldflags.

Comment (SMillerDev) [reply Formula/sqlc.rb:12]: Isn't that part of `std_go_args`?

Comment (bayandin) [reply Formula/sqlc.rb:12]: [Nope](https://github.com/Homebrew/brew/blob/2.7.0/Library/Homebrew/formula.rb#L1434-L1437):

```ruby
  # Standard parameters for Go builds.
  def std_go_args
    ["-trimpath", "-o", bin/name]
  end
```

File: Formula/sqlc.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/sqlc.rb:38]: looks good now. :)

File: Formula/sqlc.rb (hunk 3 above)
Comment (chenrui333) [reply Formula/sqlc.rb]: I wonder if we should move `ldflags -s -w` into std_go_args.

File: Formula/sqlc.rb (hunk 2 above)
Comment (kyleconroy) [reply Formula/sqlc.rb:38]: Thanks!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@kyleconroy, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**